circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/Replica.java

```diff
@@ -445,4 +437,23 @@ public TableAndStatistics getTableAndStatistics(TableReplication tableReplicatio
         tableReplication.getReplicaTableName());
   }
 
+  public void checkIfReplicaCleanupRequired(
+      String replicaDatabaseName,
+      String replicaTableName,
+      DataManipulationClient dataManipulationClient) {
+    try (CloseableMetaStoreClient client = getMetaStoreClientSupplier().get()) {
```

Comment:
Wouldn't it be better to move this to line 448? There's no need to create a client unless we get that far.